nav2_collision_monitor/params/collision_monitor_params.yaml

```diff
@@ -42,11 +42,42 @@ collision_monitor:
     FootprintApproach:
       type: "polygon"
       action_type: "approach"
-      footprint_topic: "/local_costmap/published_footprint"
+      # footprint_topic: "/local_costmap/published_footprint"
       time_before_collision: 2.0
       simulation_time_step: 0.1
       min_points: 6
-      visualize: False
+      visualize: True
+      polygon_pub_topic: "polygon_approach"
+      velocity_polygons: ["rotation", "translation_forward", "translation_backward", "stop"] # third priority
```

Comment:
Why would this be under approach footprint? this doesn't make sense

Reply:
This feature is available for all action type. The logic is [here](https://github.com/kaichie/navigation2/blob/b97cbb7a490de8c928650ebf74e5cb3f104b1ce5/nav2_collision_monitor/src/polygon.cpp#L470), if `polygon_sub_topic`(`STOP`, `SLOWDOWN`, `LIMIT`, `DO_NOTHING`) or `footprint_topic`(`APPROACH`) is not available, it will try to use the velocity polygon feature. This follows the pattern of the original design.
For real world use case, we do use the dynamic update of polygon via the `footprint_topic` for `APPROACH` model.

Reply:
Lets make the demo file of this based on the stop zones, that's how I believe most users will be using it. 

It doesn't seem good form to me to do this in the approach style. The approach takes into account the **actual** footprint, so there's no need to artificially inflate or adjust it based on the velocity profile. Its already only projecting in the velocity directions. I would not recommend that you use it that way either.